src/greynirseq/nicenlp/utils/label_schema/label_schema.py

```diff
@@ -0,0 +1,112 @@
+from collections import namedtuple
+import json
+
+from fairseq.data import Dictionary
+import torch
+import torch.nn.functional as F
+
+
+def parse_label_schema(path):
+    LabelSchema = namedtuple(
+        "LabelSchema",
+        [
+            "labels",
+            "group_name_to_labels",
+            "label_categories",
+            "category_to_group_names",
+            "separator",
+            "group_names",
+            "null",
+            "null_leaf",
+        ],
+    )
+    with open(path, "r") as fp:
+        j_obj = json.load(fp)
+    return LabelSchema(**j_obj)
+
+
+def label_schema_as_dictionary(label_schema):
+    label_dict = Dictionary()
+
+    labels = list(label_schema.labels)
+    assert len(labels) == len(set(labels))
+
+    for label in labels:
+        label_dict.add_symbol(label)
+
+    return label_dict
+
+
+def make_vec_idx_to_dict_idx(dictionary, labels, device="cpu", fill_value=-100):
+    vec_idx_to_dict_idx = torch.full(
+        (len(dictionary),), device=device, fill_value=fill_value, dtype=torch.long
+    )
+    for vec_idx, label in enumerate(labels):
+        vec_idx_to_dict_idx[vec_idx] = dictionary.index(label)
+    return vec_idx_to_dict_idx
+
+
+def make_group_masks(dictionary, schema, device="cpu"):
+    num_groups = len(schema.group_names)
+    offset = dictionary.nspecial
+    num_labels = len(dictionary) - offset
+    ret_mask = torch.zeros(num_labels, num_groups, dtype=torch.int64, device=device)
+    for cat, cat_group_names in schema.category_to_group_names.items():
+        cat_label_idx = dictionary.index(cat)
+        cat_vec_idx = schema.label_categories.index(cat)
+        for group_name in cat_group_names:
+            ret_mask[cat_vec_idx, schema.group_names.index(group_name)] = 1
+        assert cat_label_idx != dictionary.unk()
+    for cat in schema.label_categories:
+        cat_label_idx = dictionary.index(cat)
+        assert cat_label_idx != dictionary.unk()
+    return ret_mask
+
+
+def make_mapped_group_masks(schema, ldict, device="cpu"):
+    # deprecated
```

Comment:
Any reason to keep it? Also the one below

Reply:
Should've been removed.